### PR TITLE
Exposed ReactNative's Platform.select function.

### PR DESCRIPTION
**Pull request by SandyChapman**

### CODE REVIEW

**Comment by MoOx** [comment]
Interested in having this!
Could you rebase/resolve conflicts please?

**Comment by SandyChapman** [comment]
@MoOx : Rebased. Please review.

**Comment by MoOx** [comment]
We need to figure out why ci doesn't want to test this pr... (like lot's of others)

File: src/platform.re [hunk 1]
```diff
@@ -51,3 +51,25 @@ let version = () =>
...
   };
+
+type selectOptions('a) = {
+  ios: 'a,
+  android: 'a,
+};
+
+[@bs.deriving abstract]
+type _selectOptions('a) = {
+  [@bs.as "ios"]
```
Comment (czystyl) [inline src/platform.re]: I think that we don't need to annotate fields if we don't want to change the fields name

Comment (SandyChapman) [reply src/platform.re]: Good point. Will remove the annotations.

Comment (SandyChapman) [reply src/platform.re]: @czystyl : Updated. Please re-review.

**Comment by MoOx** [comment]
I am having second thoughts about this binding. Please take a look to 
https://github.com/reasonml-community/bs-react-native/issues/316

**Pull request merged**